app/models/contact_method.rb

```diff
@@ -1,12 +1,12 @@
 class ContactMethod < ApplicationRecord
   has_many :people, inverse_of: :preferred_contact_method, foreign_key: :preferred_contact_method_id, dependent: :restrict_with_error
 
-  scope :as_filter_types, -> { enabled_public.distinct(:name) }
-  scope :autoemail, -> (boolean){ where( boolean ? arel_table[:name].lower.eq("autoemail") : arel_table[:name].lower.not_eq("autoemail") ) }
+  scope :as_filter_types, -> { enabled.distinct(:name) }
+  scope :email, -> { find_by name: 'Email' }
   scope :enabled, -> { where enabled: true }
-  scope :enabled_public, ->{ enabled.autoemail(false) }
   scope :field_name, -> (field_name){ where arel_table[:field].lower.eq(field_name) }
   scope :method_name, -> (method_name){ where arel_table[:name].lower.eq(method_name) }
+  scope :sample_one, -> { order(Arel.sql('RANDOM()')).first }
```

Comment:
Allows for `ContactMethod.sample_one` instead of `ContactMethod.all.sample`. Unfortunately AR won't let me name it `sample` :(